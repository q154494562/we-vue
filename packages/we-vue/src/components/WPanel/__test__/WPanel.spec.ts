import { shallowMount } from '@vue/test-utils'
import Panel from '../'

describe('panel', () => {
  test('create', () => {
    const wrapper = shallowMount(Panel, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-panel')
    expect(wrapper.classes()).toContain('weui-panel')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render with title', () => {
    const wrapper = shallowMount(Panel, {
      propsData: {
        title: 'test-title',
      },
    })

    expect(wrapper.contains('.weui-panel__hd')).toBeTruthy()
    expect(wrapper.find('.weui-panel__hd').text()).toBe('test-title')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
