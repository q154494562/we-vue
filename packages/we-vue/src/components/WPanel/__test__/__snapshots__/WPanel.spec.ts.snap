// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel create 1`] = `
<div class="weui-panel weui-panel_access">
  <div class="weui-panel__bd"></div>
  <div class="weui-panel__ft"></div>
</div>
`;

exports[`panel render with title 1`] = `
<div class="weui-panel weui-panel_access">
  <div class="weui-panel__hd">test-title</div>
  <div class="weui-panel__bd"></div>
  <div class="weui-panel__ft"></div>
</div>
`;
